genie-web/src/main/java/com/netflix/genie/web/jpa/services/JpaJobPersistenceServiceImpl.java

```diff
@@ -465,6 +466,28 @@ public void saveJobSpecification(
             : Optional.empty();
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Transactional(readOnly = true)
+    public boolean isV4(
+        @NotBlank(message = "Id is missing and is required") final String id
+    ) {
+        log.debug("Requested to get v4 for job {}", id);
```

Comment:
This could be clearer, something like `Request to figure out whether job {} is a v4 job or not`

Reply:
Done